solr/core/src/java/org/apache/solr/search/MetaCacheRegenerator.java

```diff
@@ -192,38 +219,42 @@ public <K1, M1> boolean regenerateItem(
 
   @Override
   public <K1> SolrCache<K1, ?> wrap(SolrCache<K1, ?> internal) {
+    if (!enabled) {
+      // If this regenerator's not enabled, don't wrap, and we should remove
+      // ourselves from the associated cache.
+      if (internal instanceof SolrCacheBase) {
+        log.warn(
+            "configured cache regenerator for {} is disabled (perhaps because no autowarming?)",
+            internal);
+        ((SolrCacheBase) internal).regenerator = null;
```

Comment:
Just curious - is this an optimization? It seems like the only place that calls this is:
```
public SolrCache<?, ?> toExternal() {
    SolrCache<?, ?> internal = (SolrCache<?, ?>) this;
    return regenerator == null ? internal : regenerator.wrap(internal);
  }
```
So even if we don't set the regenerator to null, subsequent call will keep calling `wrap`, yet it will just return the unwrapped same instance, which is the same as setting it to null?

I just found this slightly off, as i don't expect a wrap function would mutate the state on the input param.